Challenges/tree.py

```diff
@@ -0,0 +1,47 @@
+class Node:
+    def __init__(self, val=None):
+        self.val = val
+        self.left = None
+        self.right = None
+
+
+class Tree:
+    def __init__(self, root=None):
+        self.root: Node = root
+
+    def __rec_insert(self, key, curr_root: Node = None):
+        if curr_root is None:
+            return Node(key)
+        else:
+            if curr_root.val == key:
+                return curr_root
+            elif curr_root.val < key:
+                curr_root.right = self.__rec_insert(key, curr_root.right)
+            else:
+                curr_root.left = self.__rec_insert(key, curr_root.left)
+        return curr_root
+
+    def insert(self, key):
```

Comment:
You can remove the `_rec_insert` function completely. Instead have a similar pattern to inorder using `Node(None)`